wandb/sdk/internal/sender_config.py

```diff
@@ -0,0 +1,156 @@
+            self._delete_at_path(_key_path(config_item))
+
+    def add_unset_keys(self, other_config_tree: Dict[str, Any]) -> None:
+        """Uses the given dict for any keys that aren't already set."""
+        for k, v in other_config_tree:
+            if k not in self._tree.items():
+                self._tree[k] = v
+
+    def to_backend_dict(
+        self,
+        telemetry_record: telemetry.TelemetryRecord,
+        framework: Optional[str],
+        start_time_millis: int,
+        metric_pbdicts: Sequence[Dict[int, Any]],
+    ) -> BackendConfigDict:
+        """Returns a dictionary representation expected by the backend.
+
+        The backend expects the configuration in a specific format, and the
+        config is also used to store additional metadata about the run.
+
+        Args:
+            telemetry_record: Telemetry information to insert.
+            framework: The detected framework used in the run (e.g. TensorFlow).
+            start_time_millis: The run's start time in Unix milliseconds.
+            metric_pbdicts: List of dict representations of metric protobuffers.
+        """
+        backend_dict = self._tree.copy()
+        wandb_internal = backend_dict.setdefault("_wandb", {})
+
+        ###################################################
+        # Telemetry information
+        ###################################################
+        py_version = telemetry_record.python_version
+        if py_version:
+            wandb_internal["python_version"] = py_version
+
+        cli_version = telemetry_record.cli_version
+        if cli_version:
+            wandb_internal["cli_version"] = cli_version
+
+        if framework:
+            wandb_internal["framework"] = framework
+
+        huggingface_version = telemetry_record.huggingface_version
+        if huggingface_version:
+            wandb_internal["huggingface_version"] = huggingface_version
+
+        wandb_internal["is_jupyter_run"] = telemetry_record.env.jupyter
+        wandb_internal["is_kaggle_kernel"] = telemetry_record.env.kaggle
+
+        # TODO: ?
+        wandb_internal["start_time"] = start_time_millis
+
+        # The full telemetry record. Admittedly this is redundant with some of
+        # the above, but we do it because.. uh...
+        wandb_internal["t"] = proto_util.proto_encode_to_dict(telemetry_record)
+
+        ###################################################
+        # Metrics
+        ###################################################
+        if metric_pbdicts:
+            wandb_internal["m"] = metric_pbdicts
+
+        return BackendConfigDict(
+            {
+                key: {
+                    # Configurations can be stored in a hand-written YAML file,
+                    # and users can add descriptions to their hyperparameters
+                    # there. However, we don't support a way to set descriptions
+                    # via code, so this is always None.
```

Comment:
maybe in the future we will allow it? something to consider for the rust client when we redo the config object on the user side process

Reply:
makes sense!